content/blog/2019/cve-2019-12995/index.md

```diff
@@ -0,0 +1,96 @@
+
+## Vulnerability impact and Detection
+
+The Istio proxy is vulnerable if the following two conditions are satisfied:
+* A JWT authentication policy is applied to it.
+* The JWT issuer (specified by `jwksUri`) uses RSA algorithm for signature verification (Note the bug is just triggered in this condition and have nothing to do with the RSA algorithm)
+
+If JWT policy is applied to the Istio ingress gateway, please be aware that any external user who has access to the ingress gateway could crash it with a single HTTP request.
+
+If JWT policy is applied to the sidecar only, please keep in mind it might still be vulnerable. For example, the Istio ingress gateway might forward the JWT token to the sidecar which could be a malformed JWT token that crashes the sidecar.
+
+A vulnerable Istio proxy will crash on an HTTP request with a malformed JWT token. When Istio proxy crashes, all existing connections will be disconnected immediately. The pilot-agent will restart the crashed Istio proxy automatically and it may take a few seconds to a few minutes for the restart. pilot-agent will stop restarting the Istio proxy after it crashed more than 10 times. In this case, Kubernetes will redeploy the pod (including the workload behind Istio proxy).
+
+To detect if there is any JWT authentication policy applied in your cluster, run the following command which print either of the following output:
+* Found JWT in authentication policy, **YOU ARE AFFECTED**
+* Did NOT find JWT in authentication policy, *YOU ARE NOT AFFECTED*
+
+{{< text bash >}}
+$ cat <<'EOF' | bash -
+set -e
+set -u
+set -o pipefail
+
+red=`tput setaf 1`
+green=`tput setaf 2`
+reset=`tput sgr0`
+
+echo "Checking authentication policy..."
+
+JWKS_URI=()
+JWKS_URI+=($(kubectl get policy --all-namespaces -o jsonpath='{range .items[*]}{.spec.origins[*].jwt.jwksUri}{" "}{end}'))
+JWKS_URI+=($(kubectl get meshpolicy --all-namespaces -o jsonpath='{range .items[*]}{.spec.origins[*].jwt.jwksUri}{" "}{end}'))
+if [ "${#JWKS_URI[@]}" != 0 ]; then
+  echo "${red}Found JWT in authentication policy, YOU ARE AFFECTED${reset}"
+  exit 1
+fi
+
+echo "${green}Did NOT find JWT in authentication policy, YOU ARE NOT AFFECTED${reset}"
+EOF
+{{< /text >}}
+
+## Mitigations
+
+This bug is fixed in the following versions of Istio:
+
+* For Istio 1.0.x deployments: update to a minimum version of Istio 1.0.9
+* For Istio 1.1.x deployments: update to a minimum version of Istio 1.1.10
+* For Istio 1.2.x deployments: update to a minimum version of Istio 1.2.2
+
+If you cannot immediately upgrade to one of these releases, you have the additional option of injecting a [`Lua` filter](https://github.com/istio/tools/tree/master/examples/luacheck) into older versions of Istio.   This filter has been verified to work with Istio versions 1.1.9, 1.0.8, 1.0.6, and 1.1.3.
+
+The `Lua` filter is injected *before* the Istio `jwt-auth` filter.
```

Comment:
This paragraph is completely out of place. What Lua filter, what does this have to do with the CVE?

Reply:
This line refers to the previous one.